### DISCUSSION TITLE
when dashboard calls api, request blocked due to CORS policy

### DISCUSSION THREAD

**Discussion by ponienchen** [sorry-cypress/sorry-cypress#306] [Q&A]
This happened to my own build, and I was wondering if I need to add the necessary flag to the request header to resolve it.
If so, can someone help point me to the correct file ?
If not so, any suggestions ?

**Comment by agoldis** [comment]
Hi @ponienchen!
There's an open issue https://github.com/sorry-cypress/sorry-cypress/issues/286.

**Reply by ponienchen** [reply]
how is it that the [demo](https://sorry-cypress-demo.herokuapp.com/) site does not have this problem?

**Comment by agoldis** [comment]
Not sure what causes the CORS problem. Please make sure there's no reverse proxy / gateway applying its own CORS policy. 

`apollo-server` actually configures quite an open CORS policy by default. 

It uses [cors](https://www.npmjs.com/package/cors) package and configures it with:

```ts
corsMiddleware({
  origin: '*',
})
```

i.e. it doesn't apply any CORS limitations

**References**

https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server/src/index.ts#L122

https://github.com/apollographql/apollo-server/blob/1bb051f63d67617488a18061971c9a83027b65f5/packages/apollo-server-express/src/ApolloServer.ts#L164

**Reply by ponienchen** [reply]
Thank you for the info.

**Comment by ponienchen** [comment]
I might not have built the docker images correctly.
May I ask how you build docker images ?

Currently, say I want to build the dashboard docker image.
What I do is copy the /packages/dashboard/Dockerfile to the root dir, and then issue the command `docker build . -t {image-name}`

Is that the way I am supposed to be doing ?

**Reply by agoldis** [reply]
Please refer to docker-compose.build.yml file, you'd do it from project root and specify the path to Dockerfile and context
